File: 215-kth-largest-element-in-an-array/215-kth-largest-element-in-an-array.java
class Solution {
    public int findKthLargest(int[] nums, int k) {
        int n = nums.length;
        k = n-k;
        int lo = 0;
        int hi = nums.length-1;
        while(lo < hi){
            int j = findPivot(nums,lo,hi);
            if(j < k){
                lo = j+1;
            }
            else if(j > k){
                hi = j-1;
            }
            else{
                break;
            }
        }
        return nums[k];
    }
    
    private int findPivot(int[] nums,int lo, int hi){
        int i = lo;
        int pivot = nums[hi];
        for(int j = lo; j < hi; j++){
            if(nums[j] <= pivot){
                swap(nums,i,j);
                i++;
            }
        }
        swap(nums,i,hi);
        return i;
    }
    
    private void swap(int[] nums, int i, int j){
        int temp = nums[i];
        nums[i] = nums[j];
        nums[j] = temp;
    }
}

File: 215-kth-largest-element-in-an-array/215-kth-largest-element-in-an-array.cpp
class Solution {
public:
    int findKthLargest(vector<int>& nums, int k) {
        sort(nums.begin(),nums.end());
        int n = nums.size();
        int cnt = 1;
        for(int i = n-1; i >= 0; i--){
            if(cnt == k){
                return nums[i];
            }
            else{
                cnt++;
            }
        }
        return -1;
    }
};